index/scorch/segment/zap/docvalues.go

```diff
@@ -62,37 +62,44 @@ func (di *docValueReader) curChunkNumber() uint64 {
 }
 
 func (s *SegmentBase) loadFieldDocValueReader(field string,
-	fieldDvLoc uint64) (*docValueReader, error) {
+	fieldDvLocStart, fieldDvLocEnd uint64) (*docValueReader, error) {
 	// get the docValue offset for the given fields
-	if fieldDvLoc == fieldNotUninverted {
+	if fieldDvLocStart == fieldNotUninverted {
 		return nil, fmt.Errorf("loadFieldDocValueReader: "+
 			"no docValues found for field: %s", field)
 	}
 
-	// read the number of chunks, chunk lengths
-	var offset, loc uint64
-	numChunks, read := binary.Uvarint(s.mem[fieldDvLoc : fieldDvLoc+binary.MaxVarintLen64])
-	if read <= 0 {
-		return nil, fmt.Errorf("failed to read the field "+
-			"doc values for field %s", field)
+	// read the number of chunks, and chunk offsets position
+	var numChunks, chunkOffsetsPosition uint64
+
+	if fieldDvLocEnd-fieldDvLocStart > 16 {
+		numChunks = binary.BigEndian.Uint64(s.mem[fieldDvLocEnd-8 : fieldDvLocEnd])
+		// read the length of chunk offsets
+		chunkOffsetsLen := binary.BigEndian.Uint64(s.mem[fieldDvLocEnd-16 : fieldDvLocEnd-8])
+		// estimate position of chunk offsets
```

Comment:
the word "estimate" here in the comment feels wrong, as it ought to be a pretty exact calculation

Reply:
Done.